R/simulate_design.R

```diff
@@ -128,17 +131,12 @@ simulate_single_design <- function(design, sims) {
   } else {
     sims <- check_sims(design, sims)
     results_list <- fan_out(design, sims)
-    # fan_id <- setNames(
-    #   lapply(rev(sims$n), seq),
-    #   paste0("fan_", rev(seq_len(nrow(sims))))
-    # )
-    # fan_id <- expand.grid(rev(fan_id))
-    s <- setNames(sims$n, paste0("step_", sims$end, "_draw"))
-    fan_id <- do.call(cbind.data.frame, lapply(
+    sims2    <- rbind(sims[1,], sims[-1,][sims[-1,]$n!=1,]) # preserve first row; drop all others with n = 1
+    sims2[1,1] <- 1                                         # step_1_draw always ends at 1
```

Comment:
should ` step_1_draw always ends at 1` be ` step_1_draw always begins at 1`

Reply:
Should be "ends" since end is used to identify sims with >1 simulations
Compare three cases below. First two are fine, but third causes problems
```
> DeclareDesign:::check_sims(simple_two_arm_designer(), c(3,1,1,1,1,1))
  end n
1   1 3
6   6 1
> DeclareDesign:::check_sims(simple_two_arm_designer(), c(1,3,1,1,1,1))
  end n
1   1 1
2   2 3
6   6 1
> DeclareDesign:::check_sims(simple_two_arm_designer(), c(1,1,3,1,1,1))
  end n
2   2 1
3   3 3
6   6 1
```

Reply:
ok, it was a confusing comment and indirect solution to what is actually a labeling problem.

instead, it would be clearer to write
```
sims_for_labels <- sims[union(1, which(sims$n > 1)), ] # remove rows that would yield duplicated step columns
s          <- setNames(sims_for_labels$n, paste0("step_", sims_for_labels$end, "_draw"))
names(s)[1] <- "step_1_draw" # first name is hardcoded to match bootstrap_diagnosands
```

which describes what it is doing and why.